src/indexing/test/lfs.cpp

```diff
@@ -0,0 +1,376 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+
+#include "ccf/indexing/strategies/seqnos_by_key_bucketed.h"
+#include "host/lfs_file_handler.h"
+#include "indexing/enclave_lfs_access.h"
+#include "indexing/test/common.h"
+
+#include <doctest/doctest.h>
+
+std::vector<uint8_t> read_file(const std::filesystem::path& p)
+{
+  std::ifstream f(p, std::ios::binary);
+  f.seekg(0, f.end);
+  const auto size = f.tellg();
+  f.seekg(0, f.beg);
+  std::vector<uint8_t> contents(size);
+  f.read((char*)contents.data(), contents.size());
+  f.close();
+  return contents;
+}
+
+void write_file_corrupted_at(
+  const std::filesystem::path& p,
+  size_t i,
+  const std::vector<uint8_t>& original)
+{
+  auto corrupted(original);
+  REQUIRE(i < corrupted.size());
+  corrupted[i]++;
+  std::ofstream f(p, std::ios::trunc | std::ios::binary);
+  f.write((char const*)corrupted.data(), corrupted.size());
+  f.close();
+}
+
+TEST_CASE("Basic cache" * doctest::test_suite("lfs"))
```

Comment:
First unit test, prodding the LFS APIs directly. Still a bunch of boiler-plate to create test ringbuffers. Then write some contents, read some contents, confirm that the integrity+encryption recognises corruption.